### PR TITLE
restore Activity.Current before all IIS LifeCycle events

### PR DESCRIPTION
**Pull request by chris-smith-zocdoc**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet/issues/3286

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/TelemetryHttpModule.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ private void OnExecuteRequestStep(HttpContextBase context, Action step)
     {
         // Called only on 4.7.1+ runtimes
 
-        if (context.CurrentNotification == RequestNotification.ExecuteRequestHandler && !context.IsPostNotification)
+        if (!context.IsPostNotification)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/TelemetryHttpModule.cs:88]: Let's remove this check as well and call `ActivityHelper.RestoreContextIfNeeded(context.ApplicationInstance.Context)` on every step.

I tested it locally and with this check in place `PostRequestHandlerExecute` won't be restored.

**Comment by CodeBlanch** [comment]
@chris-smith-zocdoc Would you please add a note about the change in the project CHANGELOG?

**Comment by chris-smith-zocdoc** [comment]
@CodeBlanch done

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* restore Activity.Current before all IIS LifeCycle events
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/CHANGELOG.md:5]:
```suggestion
* Restore Activity.Current before all IIS Lifecycle events
```

**Comment by utpilla** [comment]
@chris-smith-zocdoc Please update your branch with the latest changes from main branch. Maintainers do not have access to update your branch.

**Comment by chris-smith-zocdoc** [comment]
@utpilla done

**Comment by chris-smith-zocdoc** [comment]
@utpilla is it possible to cut a release which includes this change?

**Pull request merged**